### PR TITLE
debug test cases based on new pipeline jobs

### PR DESCRIPTION
**Pull request by Junefen**
- [x] debug test_package
- [x] debug test_function with esx
- [x] debug test_rhsm with esx
- [x] debug test_satellite with esx
- [x] debug test_esx
- [x] debug test_kubevirt
- [x] debug test_ahv
- [x] debug test_libvirt
- [x] debug test_hyperv

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -202,6 +202,12 @@ def satellite():
     return None
 
 
+@pytest.fixture(scope="class")
+def class_satellite_sca_disable(satellite):
```
Comment (Junefen) [inline tests/conftest.py:206]: subscription cases need sca disable, but the default is enable.
will maintain the pipeline job to initial setup sca as requirement

File: tests/conftest.py [hunk 2]
```diff
@@ -210,6 +216,12 @@ def rhsm():
     return None
 
 
+@pytest.fixture(scope="class")
+def class_rhsm_sca_disable(rhsm):
```
Comment (Junefen) [inline tests/conftest.py:220]: same as `class_satellite_sca_disable()`

File: tests/conftest.py [hunk 3]
```diff
@@ -351,8 +363,10 @@ def owner_data():
         f"Organization with id {bad_owner} could not be found",
         f"Couldn't find Organization '{bad_owner}'",
     ]
+    # the errors for null are different when virt-who host registered and unreigstered.
     owner["null_error"] = [
         "Communication with subscription manager failed",
+        "Unable to read certificate, system is not registered or you are not root",
```
Comment (Junefen) [inline tests/conftest.py:369]: the errors for null are different when virt-who host registered and unreigstered.

File: tests/function/test_config.py [hunk 4]
```diff
@@ -144,25 +146,12 @@ def test_print_in_virtwho_conf(self, virtwho, globalconf, hypervisor_data):
...
         globalconf.update("global", "debug", "True")
-        result = virtwho.run_service()
-        assert (
-            result["error"] == 0
-            and result["send"] == 0
-            and result["thread"] == 0
-            and result["debug"] is True
-            and guest_id in result["log"]
-        )
+        _, output = ssh_host.runcmd("virt-who", if_stdout=True)
```
Comment (Junefen) [inline tests/function/test_config.py:149]: could not use the runserver for print function, which will not print any mapping in rhsm log.

File: tests/function/test_config.py [hunk 5]
```diff
@@ -144,25 +146,12 @@ def test_print_in_virtwho_conf(self, virtwho, globalconf, hypervisor_data):
...
         globalconf.update("global", "debug", "False")
-        result = virtwho.run_service()
-        assert (
-            result["error"] == 0
-            and result["send"] == 0
-            and result["thread"] == 0
-            and result["debug"] is False
-            and guest_id not in result["log"]
-        )
+        _, output = ssh_host.runcmd("virt-who", if_stdout=True)
```
Comment (Junefen) [inline tests/function/test_config.py:153]: no mapping info in the stderr log, only in stdout output.

File: tests/function/test_config.py [hunk 6]
```diff
@@ -580,7 +569,7 @@ def test_debug_in_virtwho_sysconfig(self, virtwho, function_sysconfig):
         function_sysconfig.clean()
 
     @pytest.mark.tier1
-    def test_oneshot_in_virtwho_sysocnfig(self, virtwho, function_sysconfig):
+    def test_oneshot_in_virtwho_sysconfig(self, virtwho, function_sysconfig):
```
Comment (Junefen) [inline tests/function/test_config.py:572]: name issue

File: tests/hypervisor/conftest.py [hunk 7]
```diff
@@ -243,16 +254,18 @@ def ahv_assertion():
...
             "invalid": {
                 "xxx": login_error,
-                "红帽€467aa": "internal error: Unable to parse URI qemu+ssh",
-                "": "",
+                "红帽€467aa": username_password_non_ascii_error,
+                "": login_error,
             },
+            "disable": 'Required option: "username" not set',
         },
         "password": {
```
Comment (Junefen) [inline tests/hypervisor/conftest.py:262]: didn't define the value before.

File: tests/hypervisor/test_esx.py [hunk 8]
```diff
@@ -559,7 +559,7 @@ def test_username(self, function_hypervisor, virtwho, esx_assertion):
 
         # username option is null but another config is ok
         function_hypervisor.update("username", "")
-        result = virtwho.run_service()
+        result = virtwho.run_service(wait=30)
```
Comment (Junefen) [inline tests/hypervisor/test_esx.py:562]: time is short to get the whole log.

**Review by hkx303 — APPROVED** [review]
ACK, pending a small conflict

**Review by hkx303 — APPROVED** [review]
ACK

**Pull request merged**